### PR TITLE
Delete irreparable segments

### PR DESCRIPTION
**Pull request by aleitner**
What: 
Remove the irreparable segments from the repair queue if they fall under the minimum threshold

Why:
there's a condition in the repair checker where it checks for irreparable segments and just returns an error and if there's an error, it doesn't get removed from the queue. We need to make sure we appropriately remove from the queue in error conditions. The only time it really should stay in the queue is if there's a transient sort of error whichmeans during uploading/downloading the pieces for repair

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: satellite/repair/repairer/repairer.go [hunk 1]
```diff
@@ -109,13 +109,17 @@ func (service *Service) worker(ctx context.Context, seg *pb.InjuredSegment) (err
 
 	workerStartTime := time.Now().UTC()
 
-	zap.L().Info("Limiter running repair on segment", zap.Binary("segment", seg.GetPath()))
+	service.log.Info("Limiter running repair on segment", zap.Binary("segment", seg.GetPath()))
 	err = service.repairer.Repair(ctx, string(seg.GetPath()))
 	if err != nil {
+		if IrreparableError.Has(err) {
+			service.log.Error("deleting irreparable segment from the queue:", zap.Error(service.queue.Delete(ctx, seg)), zap.Binary("segment", seg.GetPath()))
```
Comment (mobyvb) [inline satellite/repair/repairer/repairer.go]: could you split this into multiple lines?

Comment (aleitner) [reply satellite/repair/repairer/repairer.go]: done

**Comment by jenlij** [comment]
Make sure your fill out the what/why portion of your PR description

File: satellite/repair/repairer/repairer.go [hunk 2]
```diff
@@ -109,13 +109,21 @@ func (service *Service) worker(ctx context.Context, seg *pb.InjuredSegment) (err
 
 	workerStartTime := time.Now().UTC()
 
-	zap.L().Info("Limiter running repair on segment", zap.Binary("segment", seg.GetPath()))
```
Comment (jenlij) [inline satellite/repair/repairer/repairer.go]: nice

File: satellite/repair/repairer/segments.go [hunk 3]
```diff
@@ -99,7 +102,7 @@ func (repairer *SegmentRepairer) Repair(ctx context.Context, path storj.Path) (e
 	// irreparable piece, we need k+1 to detect corrupted pieces
 	if int32(numHealthy) < pointer.Remote.Redundancy.MinReq+1 {
 		mon.Meter("repair_nodes_unavailable").Mark(1)
-		return Error.New("segment %v cannot be repaired: only %d healthy pieces, %d required", path, numHealthy, pointer.Remote.Redundancy.MinReq+1)
+		return Error.Wrap(IrreparableError.New("segment %v cannot be repaired: only %d healthy pieces, %d required", path, numHealthy, pointer.Remote.Redundancy.MinReq+1))
```
Comment (egonelbre) [inline satellite/repair/repairer/segments.go]: No need for the additional `Error.Wrap`

**Pull request merged**